src/librustc_metadata/encoder.rs

```diff
@@ -1213,13 +1213,23 @@ impl<'a, 'b: 'a, 'tcx: 'b> IsolatedEncoder<'a, 'b, 'tcx> {
         debug!("IsolatedEncoder::encode_info_for_closure({:?})", def_id);
         let tcx = self.tcx;
 
-        let data = ClosureData {
-            kind: tcx.closure_kind(def_id),
-            sig: self.lazy(&tcx.fn_sig(def_id)),
+        let kind = if let Some(sig) = self.tcx.generator_sig(def_id) {
```

Comment:
Can you add a test for cross-crate generator use to make sure there's no bug here?